cmake/EthOptions.cmake

```diff
@@ -1,6 +1,6 @@
 macro(configure_project)
 	# Default to RelWithDebInfo configuration if no configuration is explicitly specified.
-	set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Build type on single-configuration generators")
+	set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Build type on single-configuration generators" FORCE)
```

Comment:
This way it will be always `RelWithDebInfo`. We need to check for empty value first.

```
if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Build type on single-configuration generators" FORCE)
endif()

The `AND NOT CMAKE_CONFIGURATION_TYPES` added to ignore Visual Studio where CMAKE_BUILD_TYPE make no sense.